packages/lib/file-api-driver-local.ts

```diff
@@ -13,11 +13,39 @@ const { basicDelta } = require('./file-api');
 // check that it is indeed the problem, check log-database.txt of both clients, search for the note ID, and most likely both notes
 // will have been modified at the same exact second at some point. If not, it's another bug that needs to be investigated.
 
-class FileApiDriverLocal {
-	fsErrorToJsError_(error, path = null) {
+// this is the format that fs-driver-rn would return
+type StdStatResult = {
+	birthtime: Date; // Created date
+	mtime: Date; // Last modified date
+	isDirectory: ()=> boolean; // Is the file a directory?
+	path: string;
+	size: number; // Size in bytes
+};
+
+type MetaData = {
+	path: string;
+	/** in milliseconds */
+	updated_time: number;
+	isDir: boolean;
+};
+
+export type FileOptions = {encoding?: string} & (
+	| {
+		source: 'file';
+		path: string;
+	}| {
+		target: 'file';
+		path: string;
+	}
+);
+
+export class FileApiDriverLocal {
+	static fsDriver_: any;
+
+	fsErrorToJsError_(error: Error & {code?: string} , path: string | null = null) {
 		let msg = error.toString();
 		if (path !== null) msg += `. Path: ${path}`;
-		const output = new Error(msg);
+		const output = new Error(msg) as Error & {code?: string};
```

Comment:
no type please

Reply:
if I don't give it a type, I get an error when trying to set `code` property of Error object